packages/lucide-preact/src/createLucideIcon.ts

```diff
@@ -5,26 +5,32 @@ import type { IconNode, LucideIcon, LucideProps } from './types';
 
 /**
  * Create a Lucide icon component
- * @param {string} iconName
+ * @param {string} componentName
  * @param {array} iconNode
+ * @param {string} iconName
  * @returns {FunctionComponent} LucideIcon
  */
-const createLucideIcon = (iconName: string, iconNode: IconNode): LucideIcon => {
+const createLucideIcon = (
+  componentName: string,
+  iconNode: IconNode,
+  iconName: string,
+): LucideIcon => {
   const Component = ({ class: classes = '', children, ...props }: LucideProps) =>
     h(
       Icon,
       {
         ...props,
         iconNode,
         class: mergeClasses<string | JSX.SignalLike<string | undefined>>(
-          `lucide-${toKebabCase(iconName)}`,
+          `lucide-${toKebabCase(componentName)}`,
+          `lucide-${iconName}`,
```

Comment:
Alternatively we could make it, so the first argument is the actual iconName in kebabCase.
```suggestion
          `lucide-${toKebabCase(toPascalCase(iconName))}`,
          `lucide-${toKebabCase(iconName)}`
```